projection/channel/view/channels.go

```diff
@@ -0,0 +1,338 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/crypto-com/chain-indexing/appinterface/pagination"
+	"github.com/crypto-com/chain-indexing/appinterface/projection/view"
+	"github.com/crypto-com/chain-indexing/internal/utctime"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/crypto-com/chain-indexing/appinterface/rdb"
+	_ "github.com/crypto-com/chain-indexing/test/factory"
+)
+
+// Channels projection view implemented by relational database
+type Channels struct {
+	rdb *rdb.Handle
+}
+
+func NewChannels(handle *rdb.Handle) *Channels {
+	return &Channels{
+		handle,
+	}
+}
+
+func (channelsView *Channels) Insert(channel *ChannelRow) error {
+	sql, sqlArgs, err := channelsView.rdb.StmtBuilder.
+		Insert(
+			"view_channels",
+		).
+		Columns(
+			"channel_id",
+			"port_id",
+			"connection_id",
+			"counterparty_channel_id",
+			"counterparty_port_id",
+			"packet_ordering",
+			"last_in_packet_sequence",
+			"last_out_packet_sequence",
+			"total_transfer_in_count",
+			"total_transfer_out_count",
+			"total_transfer_out_success_count",
+			"total_transfer_out_success_rate",
+			"last_activity_time",
+			"bonded_tokens",
+		).
+		Values(
+			channel.ChannelID,
+			channel.PortID,
+			channel.ConnectionID,
+			channel.CounterpartyChannelID,
+			channel.CounterpartyPortID,
+			channel.PacketOrdering,
+			channel.LastInPacketSequence,
+			channel.LastOutPacketSequence,
+			channel.TotalTransferInCount,
+			channel.TotalTransferOutCount,
+			channel.TotalTransferOutSuccessCount,
+			channel.TotalTransferOutSuccessRate,
+			channel.LastActivityTime,
+			channel.BondedTokens,
```

Comment:
Do you need to marshal to json string first?

Reply:
Good catch! Let me update it!